src/arweave/composite-client.ts

```diff
@@ -514,27 +514,38 @@ export class ArweaveCompositeClient
     }
   }
 
-  async peerGetTx(url: string) {
-    const peersToTry = Array.from(this.preferredPeers);
-    const randomPeer =
-      peersToTry[Math.floor(Math.random() * peersToTry.length)];
+  async peerGetTx(url: string, retryCount = 3) {
+    const peersToTry = this.selectPeers(retryCount);
 
-    return axios({
-      method: 'GET',
-      url,
-      baseURL: randomPeer.url,
-      timeout: 500,
-    }).then(async (response) => {
-      const tx = this.arweave.transactions.fromRaw(response.data);
-      const isValid = await this.arweave.transactions.verify(tx);
-      if (!isValid) {
-        throw new Error('Invalid peer fetched transaction');
+    for (const peerUrl of peersToTry) {
```

Comment:
I think we should use `Promise.any` here since TX header fetching is somewhat performance sensitive.